src/devhostagent.restorationjob/RestorationJobApp.cs

```diff
@@ -142,10 +144,10 @@ private async Task<ExitCode> ExecuteInnerAsync(CancellationToken cancellationTok
                             perfLogger.SetProperty(HasConnectedClients, false);
 
                             var disconnectedTimeSpan = DateTimeOffset.Now - lastPingWithSessions;
-                            if (disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout)
+                            if ((disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout) || (lastPingWithSessions == null && count > 2))

```

Comment:
nit: count is generic name, it maybe better timesLastPingWithSessionsIsNull? or something like that.

Reply:
replaced count with ```timeUntilLastPingIsNull```